library/pk_wrap.h

```diff
@@ -131,6 +131,14 @@ int MBEDTLS_DEPRECATED mbedtls_pk_error_from_psa_ecdsa(psa_status_t status);
 #endif
 #endif
 
+#if defined(MBEDTLS_RSA_C)
+int mbedtls_pk_psa_rsa_sign_ext(psa_algorithm_t psa_alg_md,
+                                mbedtls_rsa_context *rsa_ctx,
+                                const unsigned char *hash, size_t hash_len,
+                                unsigned char *sig, size_t sig_size,
+                                size_t *sig_len);
+#endif /* MBEDTLS_RSA_C */
+
 #endif /* MBEDTLS_USE_PSA_CRYPTO */
 
 #if defined(MBEDTLS_PSA_CRYPTO_C)
```

Comment:
Question: is this really still `PSA_CRYPTO_C` or should it become `USE_PSA_CRYPTO` now?

The commit message says "The last references to MBDETLS_PSA_CRYPTO_C are removed from pk.[ch]." which I like, but perhaps we can go further and cover `pk_wrap.[ch]` as well?

Reply:
Err, this is actually not true anymore, as I had do undo some of these changes that were too hopeful in this regard. I will edit the commit message.

It seems like `pk_wrap` could be covered as well if it is agreed that `mbedtls_pk_error_from_psa*()` are only needed under `USE_PSA_CRYPTO`.
Actually, a quick grep reveals that nothing is using these functions. What's more, they are marked as deprecated.
`pk_wrap` being located in `library/`, does this mean that it's to be used only within Mbed TLS? If so, these functions could be removed entirely?

Reply:
Ah, excellent point! Normally the deprecation process is only for public functions, I'm not sure why we didn't remove those functions immediately - perhaps that was a temporary thing and we forgot to come back to it. Anyway, please feel free to remove them!

I had a quick look at other remaining uses of `MBEDTLS_PSA_CRYPTO_C`, and found that in `pk.h` in `mbedtls_pk_contex_t` the field `priv_id` is guarded by `MBEDTLS_PSA_CRYPTO_C` (with a comment that mentions the `sign_ext()` thing btw). IMO that should be changed to `USE_PSA_CRYPTO` as well: opaque keys are only available when `USE_PSA_CRYPTO` is enabled. The comment should be updated as follows:
- remove the 2nd note as `sign_ext()` is no longer a special case (thanks to you!)
- s/PSA_CRYPTO_C/USE_PSA_CRYPTO/ in the first bullet point

Once that's done (as well as the repercussions in files using that field), I think we can get to a point when `pk.*` and `pk_wrap.*` indeed never refer to `PSA_CRYPTO_C`, only `USE_PSA_CRYPTO` - as they should.

Reply:
Alright, I proceeded to doing what you said, let's see whether the CI has any objection!